doc/onboarding_guide.md

```diff
@@ -0,0 +1,95 @@
+Onboarding Best Practices
+=========================
+
+Reach out to `azpycli@microsoft.com` to get the process started. You'll be assigned a dev contact on the CLI team. Early and frequent communication with this contact is essential to ensuring a smooth onboarding.
+
+##Extension vs. Module
+
+One of the key decisions you will need to make is whether to create your commands in a CLI module or an extension.
+
+####Extensions
+
+|                      PROS                      |                         CONS                         |
+|:----------------------------------------------:|:----------------------------------------------------:|
+| Release separate from the CLI release schedule | Requires dedicated installation (az extension add …) |
+| Higher velocity fixes possible                 | Can be broken by changes to the azure-cli-core       |
+| Experimental UX is permissible                 |                                                      |
+
+####CLI Modules
+
+|                                   PROS                                  |                                                                             CONS                                                                             |
+|:-----------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| Comes automatically with the CLI. No dedicated installation required.   | Strictly tied to CLI release schedule                                                                                                                        |
```

Comment:
maybe keep the strictly all caps consistent between both lines?

Reply:
I don't understand your comment.

Reply:
First row says "**Strictly** tied to CLI release schedule"
second says "**STRICTLY** tied to CLI authoring guidelines" 

Was just saying the capitalization is not consistent.

Reply:
Oh, I see. I wanted more emphasis the second time. :)

Reply:
ok! I mean heck you could even make it bold. I'll be all for it!